CMakeLists.txt

```diff
@@ -49,6 +54,11 @@ add_compile_definitions(REALM_ENABLE_ENCRYPTION)
 add_compile_definitions(REALM_INSTALL_LIBEXECDIR)
 add_compile_definitions(REALM_BUILD_LIB_ONLY)
 
+if(NOT APPLE AND NOT WINDOWS_STORE AND NOT ANDROID AND VCPKG_TOOLCHAIN)
+    set(REALM_HAVE_UV 1)
+    add_compile_definitions(REALM_HAVE_UV=1)
```

Comment:
Should we be setting this here? There's the whole libuv-discovery logic a little further down. You could just hoist it above the `add_subdirectory(src)` call to make sure the scheduler macros are defined ahead of time.